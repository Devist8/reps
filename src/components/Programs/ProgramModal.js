import React from "react";

//MUI
import { makeStyles } from "@material-ui/core/styles";
import {
    Card,
    CardMedia,
    CardContent,
    CardActions,
    CardActionArea,
    Grid,
    Typography,
    TextField,
    Button,
    Slide,
} from "@material-ui/core";

//Components
import { Difficulty } from "../Difficulty";
import { Workout } from "../Workouts/Workout";

const useStyles = makeStyles((theme) => ({
    root: {
        width: "50rem",
        overflow: "hidden",
    },
    content: {
        backgroundColor: theme.palette.primary.main,
    },
    button: {
        margin: "auto 0 0 2rem",
        marginTop: "2rem",
        marginRight: "2rem",
        color: "#000",
        justifyContent: "bottom",
        alignItems: "bottom",
    },
    stats: {
        marginLeft: "1.8rem",
        textAlign: "right",
    },
    workoutListContainer: {
        backgroundColor: theme.palette.secondary.main,
        overflow: "hidden",
    },
    weekTitle: {
        backgroundColor: theme.palette.primary.light,
        overflow: "hidden",
    },
    bottomBorder: {
        height: "25px",
        backgroundColor: theme.palette.primary.main,
        borderRadius: "0 4px",
    },
    workoutContainer: {
        flexWrap: "wrap",
        flexDirection: "row",
        display: "flex",
    },
}));

export const WorkoutList = (props) => {
    const classes = useStyles();
    const { workouts } = props;
    const [openIndex, setOpenIndex] = React.useState([]);
    const handleOpen = (e) => {
        e.persist();
        setOpenIndex((prevState) => {
            if (prevState.includes(e.target.selectedIndex)) {
                const selected = prevState.indexOf(e.target.selectedIndex);
                const tempArray = prevState
                    .slice(0, selected)
                    .concat(prevState.slice(selected + 1));
                console.log(tempArray);
                console.log("remove");
                return tempArray;
            } else {
                console.log("add");
                return prevState.concat(e.target.selectedIndex);
            }
        });
    };
    console.log(openIndex);
    const orderedWorkouts = Object.keys(workouts)
        .sort()
        .reduce((obj, key) => {
            obj[key] = workouts[key];
            return obj;
        }, {});
    console.log(orderedWorkouts);
    return (
        <Grid container className={classes.workoutListContainer}>
            {Object.entries(orderedWorkouts).map((week, weekIndex) => {
                return (
                    <Grid container style={{ overflow: "hidden" }}>
                        <Grid item xs={12} className={classes.weekTitle}>
                            <CardActionArea
                                onClick={(e) => {
                                    e.target.selectedIndex = weekIndex;
                                    handleOpen(e);
                                }}
                            >
                                <Typography>{week[0]}</Typography>
                            </CardActionArea>
                        </Grid>
                        <Slide
                            in={openIndex.includes(weekIndex)}
                            mountOnEnter
                            unmountOnExit
                            timeout={{
                                enter: week[1].length * 200 - weekIndex * 200,
                                exit: 200 + weekIndex * 200,
                            }}
                        >
                            <Grid
                                item
                                xs={12}
                                className={classes.workoutContainer}
                            >
                                {week[1].map((workout, index) => {
                                    return (
                                        <Slide
                                            in={openIndex.includes(weekIndex)}
                                            mountOnEnter
                                            unmountOnExit
                                            timeout={{
                                                enter:
                                                    workout.exercises.length *
                                                        350 -
                                                    index * 350,
                                                exit: 250 + index * 250,
                                            }}
                                            style={{ overflow: "hidden" }}
                                        >
                                            <Grid
                                                item
                                                xs={6}
                                                style={{ overflow: "hidden" }}
                                            >
                                                <Workout workout={workout} />
                                            </Grid>
                                        </Slide>
                                    );
                                })}
                            </Grid>
                        </Slide>
                    </Grid>
                );
            })}
        </Grid>
    );
};

export const ProgramModal = (props) => {
    const classes = useStyles();
    const { program, edit } = props;

    return (
        <Grid
            container
            className={classes.root}
            data-testid="program-modal-test"
        >
            <Card elevation={2} style={{}}>
                <Grid item xs={12} style={{ height: "200px" }}>
                    <CardMedia
                        component="img"
                        src="/beachbody-original.jpg"
                        style={{ objectFit: "fill", height: "200px" }}
                    />
                </Grid>
                <CardContent className={classes.content}>
                    <Grid container>
                        <Grid item xs={9}>
                            {!edit ? (
                                <Typography
                                    variant="h4"
                                    style={{ fontWeight: 600 }}
                                >
                                    {program.title}
                                </Typography>
                            ) : (
                                <TextField
                                    name="title"
                                    id="title"
                                    aria-label={program.title}
                                    aria-required="true"
                                    value={program.title}
                                />
                            )}

                            <Difficulty difficulty={program.difficulty} />

                            <Typography
                                style={{
                                    marginTop: "1rem",
                                    fontWeight: 700,
                                }}
                                variant="h6"
                            >
                                Description
                            </Typography>
                            {!edit ? (
                                <Typography>{program.description}</Typography>
                            ) : (
                                <TextField
                                    name="description"
                                    value={program.description}
                                    fullWidth
                                    multiline
                                    aria-multiline="true"
                                />
                            )}
                        </Grid>
                        <Grid item xs={3}>
                            <Typography className={classes.stats}>
                                5 weeks
                            </Typography>
                            <Typography className={classes.stats}>
                                20 workouts
                            </Typography>
                            <Typography className={classes.stats}>
                                50 exercises
                            </Typography>
                            <Button
                                variant="contained"
                                color="secondary"
                                className={classes.button}
                            >
                                Add Program
                            </Button>
                        </Grid>
                    </Grid>
                </CardContent>

                <WorkoutList workouts={program.workouts} />
                <div className={classes.bottomBorder} />
            </Card>
        </Grid>
    );
};
