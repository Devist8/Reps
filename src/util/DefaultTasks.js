const defaultTasks = [
  {
    activity: 'Push-ups',
    repXP: 20,
    reps: '2 x 15',
    duration: 30,
    difficulty: 'Beginner',
    musclesTargeted: ['chest', 'biceps'],
    repId: 'zqtwitxyJ2uczAUAgtU5',
    createdAt: '2020-10-04T01:47:44.816Z',
    complete: false,
    taskType: 'Rep challenge',
    sender: 'Teddy',
    caloriesBurned: 125,
  },
  {
    activity: 'Sit-ups',
    repXP: 30,
    reps: '5 x 4',
    duration: 20,
    difficulty: 'Beginner',
    musclesTargeted: ['abs', 'legs'],
    repId: 'FWha4HurEhsEdgdKjCZh',
    createdAt: '2020-09-26T17:15:34.980Z',
    complete: false,
    taskType: 'Rep challenge',
    sender: 'Teddy',
    caloriesBurned: 125,
  },
  {
    activity: 'Squats',
    repXP: 40,
    reps: '10 x 5',
    duration: 40,
    difficulty: 'Intermediate',
    musclesTargeted: ['chest', 'biceps'],
    repId: 'yCZQ4AnUBLZjKyvHeFBH',
    createdAt: '2020-10-03T14:15:44.494Z',
    complete: false,
    taskType: 'Rep challenge',
    sender: 'Teddy',
    caloriesBurned: 125,
  },
];
