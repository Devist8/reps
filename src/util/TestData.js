export const exercise = {
    title: "Lunges",
    difficulty: "3",
    activity: "calisthenics",
    equipment: [],
    motion: "lunges, moderate effort",
    muscles: ["legs", "calfs"],
    schedule: [],
    type: "exercise",
    videoURL:
        "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/24268207545.mp4?alt=media&token=a7649cc6-e5ea-4359-93fd-d2f1dfe7dd5c",
};

export const workout = {
    title: "First Workout",
    imageURL: "/beachbody.jpg",
    description:
        "First workout to populate the UI & database but I need a bigger description for edge case purposes. You get it. Just hold on tight while I draw up this text really quick so that I can see what a few lines of text does the height of the content container.",
    difficulty: "1",
    type: "workout",
    equipment: [],
    exerciseCount: 2,
    exercises: [
        {
            title: "Lunges",
            activity: "calisthenics",
            motion: "lunges, moderate effort",
            docId: "f1b4e65b-9004-4213-8e6a-b36f83311eb1",
            difficulty: "3",
            duration: "10",
            muscles: ["legs", "calfs"],
            reps: "2x3",
            videoURL:
                "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/24268207545.mp4?alt=media&token=a7649cc6-e5ea-4359-93fd-d2f1dfe7dd5c",
            type: "exercise",
            status: 0,
            schedule: [],
            referenceId: "MUATNgInAKho3nmIqq5A",
        },
        {
            title: "Push ups",
            activity: "calisthenics",
            difficulty: "2.5",
            docId: "a73b4a82-1b29-4519-aca7-2024a52d877c",
            duration: "10",
            motion: "push ups, moderate effort",
            muscles: ["biceps", "triceps", "chest"],
            equipment: [],
            referenceId: "VPNiOW4mtq0VUfmBAOBB",
            reps: "10x2",
            type: "exercise",
            videoURL: "",
        },
    ],
};

//Program data
export const program = {
    imageURL:
        "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/43857746632.jpg?alt=media&token=6610f8ff-8951-4fbc-b12c-caea14037000",
    equipment: [],
    workouts: {
        "Week 1": [
            {
                description: "Second workout to populate the UI & database",
                title: "Second Workout",
                type: "workout",
                difficulty: "1",
                duration: "0",
                exerciseCount: 3,
                muscles: [],
                schedule: [],
                docId: "4BDnV1Pm19c2eRhh2y1h",
                imageURL: "/beachbody.jpg",
                equipment: [],
                exercises: [
                    {
                        docId: "f1b4e65b-9004-4213-8e6a-b36f83311eb1",
                        startTime: "",
                        schedule: [],
                        activity: "calisthenics",
                        type: "exercise",
                        status: 0,
                        equipment: [],
                        muscles: ["legs", "calfs"],
                        reps: "2x3",
                        referenceId: "MUATNgInAKho3nmIqq5A",
                        difficulty: "3",
                        motion: "lunges, moderate effort",
                        title: "Lunges",
                        duration: "10",
                        videoURL:
                            "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/24268207545.mp4?alt=media&token=a7649cc6-e5ea-4359-93fd-d2f1dfe7dd5c",
                    },
                    {
                        type: "exercise",
                        title: "Push ups",
                        activity: "calisthenics",
                        difficulty: "2.5",
                        status: 0,
                        muscles: ["biceps", "triceps", "chest"],
                        motion: "push ups, moderate effort",
                        startTime: "",
                        docId: "a73b4a82-1b29-4519-aca7-2024a52d877c",
                        schedule: [],
                        reps: "10x2",
                        referenceId: "VPNiOW4mtq0VUfmBAOBB",
                        videoURL: "",
                        equipment: [],
                        duration: "10",
                    },
                    {
                        difficulty: "2.5",
                        activity: "calisthenics",
                        type: "exercise",
                        title: "Sit ups",
                        startTime: "",
                        referenceId: "tJCpH4ZDTmuL8saZGqfR",
                        duration: "10",
                        equipment: [],
                        schedule: [],
                        muscles: ["abs"],
                        reps: "5x3",
                        status: 0,
                        motion: "sit ups, light effort",
                        videoURL: "",
                        docId: "67fd4cb7-e49e-4641-bd04-8947ac35c68b",
                    },
                ],
            },
            {
                type: "workout",
                exerciseCount: 2,
                exercises: [
                    {
                        duration: "5",
                        activity: "conditioning",
                        motion: "home exercise, general",
                        title: "Test 2",
                        reps: "1x3",
                        equipment: [],
                        referenceId: "T8Y9tKIPcR7IqySCfC2w",
                        type: "exercise",
                        status: 0,
                        startTime: "",
                        videoURL: "",
                        docId: "80eb89cf-250d-456c-b648-f50f8fd02753",
                        muscles: ["obliques", "abs", "legs"],
                        difficulty: "3",
                        schedule: [],
                    },
                    {
                        motion: "lunges, vigorous effort",
                        activity: "calisthenics",
                        status: 0,
                        reps: "2x5",
                        startTime: "",
                        equipment: [],
                        schedule: [],
                        docId: "fa532826-fdf8-4dac-b346-d48df0915b91",
                        duration: "5",
                        title: "Test exercise",
                        muscles: ["legs"],
                        referenceId: "WrfR2Cv2Prss1bd1Sfk0",
                        videoURL: "",
                        type: "exercise",
                        difficulty: "3",
                    },
                ],
                difficulty: "1",
                duration: "0",
                title: "Third workout",
                description: "Third workout",
                schedule: [],
                muscles: [],
                docId: "OZQxgKhTOMVs8mf7zBLr",
                equipment: [],
                imageURL: "/beachbody.jpg",
            },
            {
                schedule: [],
                docId: "ZGH6FQIcq6RCKI16OUwn",
                muscles: [],
                title: "First Workout",
                exercises: [
                    {
                        reps: "2x3",
                        startTime: "",
                        type: "exercise",
                        status: 0,
                        muscles: ["legs", "calfs"],
                        schedule: [],
                        equipment: [],
                        difficulty: "3",
                        referenceId: "MUATNgInAKho3nmIqq5A",
                        motion: "lunges, moderate effort",
                        activity: "calisthenics",
                        duration: "10",
                        videoURL:
                            "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/24268207545.mp4?alt=media&token=a7649cc6-e5ea-4359-93fd-d2f1dfe7dd5c",
                        docId: "f1b4e65b-9004-4213-8e6a-b36f83311eb1",
                        title: "Lunges",
                    },
                    {
                        activity: "calisthenics",
                        startTime: "",
                        docId: "a73b4a82-1b29-4519-aca7-2024a52d877c",
                        title: "Push ups",
                        equipment: [],
                        type: "exercise",
                        status: 0,
                        reps: "10x2",
                        difficulty: "2.5",
                        videoURL: "",
                        duration: "10",
                        referenceId: "VPNiOW4mtq0VUfmBAOBB",
                        schedule: [],
                        muscles: ["biceps", "triceps", "chest"],
                        motion: "push ups, moderate effort",
                    },
                ],
                exerciseCount: 2,
                duration: "0",
                type: "workout",
                imageURL: "/beachbody.jpg",
                description: "First workout to populate the UI & database",
                difficulty: "1",
                equipment: [],
            },
        ],
        "Week 2": [
            {
                equipment: [],
                difficulty: "1",
                description: "Third workout",
                title: "Third workout",
                duration: "0",
                docId: "OZQxgKhTOMVs8mf7zBLr",
                imageURL: "/beachbody.jpg",
                schedule: [],
                exerciseCount: 2,
                type: "workout",
                exercises: [
                    {
                        schedule: [],
                        status: 0,
                        reps: "1x3",
                        title: "Test 2",
                        equipment: [],
                        docId: "80eb89cf-250d-456c-b648-f50f8fd02753",
                        startTime: "",
                        type: "exercise",
                        videoURL: "",
                        muscles: ["obliques", "abs", "legs"],
                        difficulty: "3",
                        referenceId: "T8Y9tKIPcR7IqySCfC2w",
                        duration: "5",
                        motion: "home exercise, general",
                        activity: "conditioning",
                    },
                    {
                        activity: "calisthenics",
                        equipment: [],
                        referenceId: "WrfR2Cv2Prss1bd1Sfk0",
                        videoURL: "",
                        type: "exercise",
                        duration: "5",
                        schedule: [],
                        reps: "2x5",
                        title: "Test exercise",
                        startTime: "",
                        difficulty: "3",
                        muscles: ["legs"],
                        docId: "fa532826-fdf8-4dac-b346-d48df0915b91",
                        status: 0,
                        motion: "lunges, vigorous effort",
                    },
                ],
                muscles: [],
            },
            {
                imageURL: "/beachbody.jpg",
                muscles: [],
                description: "Second workout to populate the UI & database",
                difficulty: "1",
                schedule: [],
                type: "workout",
                exerciseCount: 3,
                equipment: [],
                duration: "0",
                exercises: [
                    {
                        muscles: ["legs", "calfs"],
                        duration: "10",
                        schedule: [],
                        equipment: [],
                        reps: "2x3",
                        startTime: "",
                        videoURL:
                            "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/24268207545.mp4?alt=media&token=a7649cc6-e5ea-4359-93fd-d2f1dfe7dd5c",
                        motion: "lunges, moderate effort",
                        difficulty: "3",
                        activity: "calisthenics",
                        referenceId: "MUATNgInAKho3nmIqq5A",
                        status: 0,
                        title: "Lunges",
                        docId: "f1b4e65b-9004-4213-8e6a-b36f83311eb1",
                        type: "exercise",
                    },
                    {
                        reps: "10x2",
                        duration: "10",
                        muscles: ["biceps", "triceps", "chest"],
                        startTime: "",
                        equipment: [],
                        referenceId: "VPNiOW4mtq0VUfmBAOBB",
                        type: "exercise",
                        videoURL: "",
                        difficulty: "2.5",
                        schedule: [],
                        title: "Push ups",
                        activity: "calisthenics",
                        status: 0,
                        motion: "push ups, moderate effort",
                        docId: "a73b4a82-1b29-4519-aca7-2024a52d877c",
                    },
                    {
                        duration: "10",
                        equipment: [],
                        videoURL: "",
                        motion: "sit ups, light effort",
                        startTime: "",
                        reps: "5x3",
                        difficulty: "2.5",
                        referenceId: "tJCpH4ZDTmuL8saZGqfR",
                        activity: "calisthenics",
                        title: "Sit ups",
                        muscles: ["abs"],
                        docId: "67fd4cb7-e49e-4641-bd04-8947ac35c68b",
                        status: 0,
                        type: "exercise",
                        schedule: [],
                    },
                ],
                docId: "4BDnV1Pm19c2eRhh2y1h",
                title: "Second Workout",
            },
            {
                description: "First workout to populate the UI & database",
                equipment: [],
                title: "First Workout",
                schedule: [],
                duration: "0",
                exercises: [
                    {
                        type: "exercise",
                        status: 0,
                        duration: "10",
                        docId: "f1b4e65b-9004-4213-8e6a-b36f83311eb1",
                        activity: "calisthenics",
                        schedule: [],
                        videoURL:
                            "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/24268207545.mp4?alt=media&token=a7649cc6-e5ea-4359-93fd-d2f1dfe7dd5c",
                        title: "Lunges",
                        motion: "lunges, moderate effort",
                        startTime: "",
                        referenceId: "MUATNgInAKho3nmIqq5A",
                        difficulty: "3",
                        equipment: [],
                        reps: "2x3",
                        muscles: ["legs", "calfs"],
                    },
                    {
                        muscles: ["biceps", "triceps", "chest"],
                        duration: "10",
                        type: "exercise",
                        schedule: [],
                        equipment: [],
                        startTime: "",
                        videoURL: "",
                        activity: "calisthenics",
                        status: 0,
                        docId: "a73b4a82-1b29-4519-aca7-2024a52d877c",
                        title: "Push ups",
                        reps: "10x2",
                        referenceId: "VPNiOW4mtq0VUfmBAOBB",
                        difficulty: "2.5",
                        motion: "push ups, moderate effort",
                    },
                ],
                type: "workout",
                exerciseCount: 2,
                muscles: [],
                docId: "ZGH6FQIcq6RCKI16OUwn",
                imageURL: "/beachbody.jpg",
                difficulty: "1",
            },
        ],
    },
    type: "program",
    description:
        "First program to add to my database but I need a bigger description for edge case purposes. You get it. Just hold on tight while I draw up this text really quick so that I can see what a few lines of text does the height of the content containerbut I need a bigger description for edge case purposes. You get it. Just hold on tight while I draw up this text really quick so that I can see what a few lines of text does the height of the content container ",
    exerciseCount: 14,
    schedule: [],
    difficulty: "2",
    duration: "0",
    workoutCount: 6,
    title: "First Program",
};

export const meals = [
    {
        title: "Immunity-Boosting Smoothe",
        imageURL:
            "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/smoothie.jpg?alt=media&token=68e83da3-5d8e-4195-ab1f-89989c36d0ee",
        ingredients: [
            "4 clementines, peeled",
            "1 ripe banana, sliced",
            "1 skinny carrot, peeled and shaved",
            "1 piece fresh ginger, peeled and minced",
            "1/4 cup plain yogurt",
            "pinch of salt",
            "1 tablespoon honey",
            "1 1/2 cups ice",
        ],
        directions: [
            "Add clementines, banana, carrot, ginger, yogurt, honey, salt, and ice to belender.",
            "Blend until smooth.",
            "Serve",
            "Enjoy!",
        ],
        labels: ["Vegetarian", "Gluten Free", "Healthy choice"],
        nutrition: {
            Protein: 9,
            Fat: 2,
            Sugar: 84,
            Fiber: 13,
            Carbs: 120,
            Calories: 491,
        },
        rating: 4.7,
        type: "snack",
    },
    {
        title: "Chicken & Veggie Stir-Fry",
        imageURL:
            "https://firebasestorage.googleapis.com/v0/b/reps-699b0.appspot.com/o/chicken%26veggie.jpg?alt=media&token=2f7db04a-c475-4a43-a586-23f3872bb71e",
        ingredients: [
            "1 lb chicken breast, cubed",
            "1 lb broccoli florets",
            "salt, to taste",
            "pepper, to taste",
            "3 tablespoons oil, for frying",
            "3 cloves garlic, minced",
            "1 tablespoon ginger, minced",
            "1 cup chicken broth",
            "2 teaspoons sesame oil",
            "¼ cup flour",
            "⅓ cup reduced sodium soy sauce",
        ],
        directions: [
            "In a large pan on medium-high heat, add 1 tablespoon of oil. Once the oil is hot, add chicken, season with salt and pepper, and sauté until cooked through and browned. Remove cooked chicken from pan and set aside. In the same pan, heat 1 tablespoon of oil and add mushrooms. When the mushrooms start to soften, add broccoli florets and stir-fry until the broccoli is tender. Remove cooked mushrooms and broccoli from the pan and set aside.",
            "Add 1 tablespoon of oil to the pan and sauté garlic and ginger until fragrant. Add the remaining sauce ingredients and stir until smooth.",
            "Return the chicken and vegetables to the saucy pan, stir until heated through.",
            "Serve with hot rice or noodles.",
            "Enjoy!",
        ],
        labels: ["Healthy Choice", "Meal Prep"],
        nutrition: {
            Carbs: 23,
            Fiber: 2,
            Protein: 27,
            Calories: 393,
            Fat: 20,
            Sugar: 4,
        },
        rating: 4.7,
        type: "dinner",
    },
];
