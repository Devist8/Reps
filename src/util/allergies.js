const allergies = [
  'celery',
  'gluten',
  'crustaceans',
  'eggs',
  'fish',
  'lupin',
  'milk',
  'molluscs',
  'mustard',
  'peanuts',
  'seasame',
  'soybeans',
  'sulphur dioxide',
  'tree nuts',
  'none',
];

export default allergies;
