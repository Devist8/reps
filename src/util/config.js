import firebase from "firebase";

const firebaseApp = firebase.initializeApp({
    apiKey: "AIzaSyBLGnhWNspnEWGMZojO6eTKdUE8FpBtFg0",
    authDomain: "reps-699b0.firebaseapp.com",
    databaseURL: "https://reps-699b0.firebaseio.com",
    projectId: "reps-699b0",
    storageBucket: "reps-699b0.appspot.com",
    messagingSenderId: "468770503396",
    appId: "1:468770503396:web:3c7b5eed6f0761acc64af1",
    measurementId: "G-8N3ZVE943C",
});

const db = firebaseApp.firestore();
const auth = firebase.auth();

export { db, auth };
