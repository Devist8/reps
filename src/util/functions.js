import { useSelector, useDispatch } from "react-redux";
import firebase from "firebase/app";
import "firebase/storage";
import { SET_AUTHENTICATED } from "../redux/types";
import axios from "axios";
import dayjs from "dayjs";
import LocalizedFormat from "dayjs/plugin/localizedFormat";

export const getWorkoutCount = (program) => {
    let count = 0;
    Object.values(program.workouts).forEach((week) => {
        count += week.length;
    });
    return count;
};

export const generateDateRange = (itemToSchedule) => {
    const prefferedDays = ["0", "1", "3", "5"];
    const newDateRange = [];
    let dateRange = [...itemToSchedule.dateRange];
    let workoutCount = itemToSchedule.workoutCount
        ? itemToSchedule.workoutCount
        : getWorkoutCount(itemToSchedule);
    let currentDate = dateRange[0];

    while (workoutCount > 0) {
        if (prefferedDays.includes(dayjs(currentDate).format("d"))) {
            newDateRange.push(currentDate);
            workoutCount = workoutCount - 1;
            currentDate = currentDate + 86400000;
        } else {
            currentDate = currentDate + 86400000;
        }
    }
    return newDateRange;
};

export const scheduleExercises = (workout) => {
    workout.exercises.map((exercise, index) => {
        exercise.date = workout.date;
    });

    return workout;
};

export const sortObjsByTitle = (array) => {
    array.sort((a, b) => {
        let titleA = a.title.toUpperCase();
        let titleB = b.title.toUpperCase();

        if (titleA < titleB) {
            return -1;
        }

        if (titleA > titleB) {
            return 1;
        }

        return 0;
    });
    return array;
};

export const sortObjsByDifficulty = (array) => {
    array.sort((a, b) => {
        let difficultyA = parseFloat(a.difficulty);
        let difficultyB = parseFloat(b.difficulty);

        if (difficultyA < difficultyB) {
            return -1;
        }

        if (difficultyA > difficultyB) {
            return 1;
        }

        return 0;
    });

    return array;
};
