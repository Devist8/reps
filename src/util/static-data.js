export const nutritionItems = [
    "Fat",
    "Calories",
    "Carbs",
    "Cholesterol",
    "Protein",
    "Sugar",
    "Sodium",
    "Vitamin A",
    "Vitamin C",
    "Calcium",
    "Iron",
    "Vitamin D",
    "Niacin",
    "Potassium",
];

export const mealTypes = ["breakfast", "lunch", "dinner", "snack", "dessert"];

export const mealLabels = [
    "vegan",
    "non-dairy",
    "nut-free",
    "pescatarian",
    "vegeatarian",
    "guten-free",
    "sugar-free",
    "low-cal",
    "meal-prep",
    "healthy choice",
    "quik-n-eazy",
    "low-carb",
];

export const muscles = [
    "abs",
    "chest",
    "biceps",
    "triceps",
    "calfs",
    "glutes",
    "back",
    "shoulders",
    "legs",
    "delts",
];
export const equipment = [
    "dumbbells",
    "kettlebells",
    "resistance bands",
    "exercise ball",
    "jump rope",
    "stairs",
];
