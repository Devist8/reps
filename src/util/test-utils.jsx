import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "./redux/store";

const WithProviders = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
};

const customRender = render(ui, { wrapper: WithProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
