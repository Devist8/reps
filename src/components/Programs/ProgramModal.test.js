import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ProgramModal } from "./ProgramModal";
import { program } from "../../util/TestData";

describe("Testing render", () => {
    test("Should render workout", () => {
        const { getByTestId } = render(<ProgramModal program={program} />);
        const tree = renderer
            .create(<ProgramModal program={program} />)
            .toJSON();
        const programElement = screen.getByTestId("program-modal-test");
        expect(programElement).toBeInTheDocument();
        expect(tree).toMatchSnapshot();
        expect(programElement).toHaveTextContent("First Program");
    });
});
