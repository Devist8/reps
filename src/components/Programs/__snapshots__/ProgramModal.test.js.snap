// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render Should render workout 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
  data-testid="program-modal-test"
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation2 MuiPaper-rounded"
    style={Object {}}
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      style={
        Object {
          "height": "200px",
        }
      }
    >
      <img
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        src="/beachbody-original.jpg"
        style={
          Object {
            "height": "200px",
            "objectFit": "fill",
          }
        }
      />
    </div>
    <div
      className="MuiCardContent-root makeStyles-content-2"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4"
            style={
              Object {
                "fontWeight": 600,
              }
            }
          >
            First Program
          </h4>
          <div
            className="MuiGrid-root makeStyles-root-9 MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-iconContainer-10 MuiGrid-item"
              style={
                Object {
                  "width": "1.35rem",
                }
              }
            >
              <svg>
                Dumbbell.svg
              </svg>
            </div>
            <div
              className="MuiGrid-root makeStyles-iconContainer-10 MuiGrid-item"
              style={
                Object {
                  "width": "1.35rem",
                }
              }
            >
              <svg>
                Dumbbell.svg
              </svg>
            </div>
          </div>
          <h6
            className="MuiTypography-root MuiTypography-h6"
            style={
              Object {
                "fontWeight": 700,
                "marginTop": "1rem",
              }
            }
          >
            Description
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            First program to add to my database but I need a bigger description for edge case purposes. You get it. Just hold on tight while I draw up this text really quick so that I can see what a few lines of text does the height of the content containerbut I need a bigger description for edge case purposes. You get it. Just hold on tight while I draw up this text really quick so that I can see what a few lines of text does the height of the content container 
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <p
            className="MuiTypography-root makeStyles-stats-4 MuiTypography-body1"
          >
            5 weeks
          </p>
          <p
            className="MuiTypography-root makeStyles-stats-4 MuiTypography-body1"
          >
            20 workouts
          </p>
          <p
            className="MuiTypography-root makeStyles-stats-4 MuiTypography-body1"
          >
            50 exercises
          </p>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-containedSecondary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Add Program
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-workoutListContainer-5 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-weekTitle-6 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              Week 1
            </p>
            <span
              className="MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-weekTitle-6 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              Week 2
            </p>
            <span
              className="MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-bottomBorder-7"
    />
  </div>
</div>
`;
